src/main/scala/org/scalacheck/Properties.scala

```diff
@@ -46,7 +47,10 @@ class Properties(val name: String) {
 
   /** Returns all properties of this collection in a list of name/property
    *  pairs.  */
-  def properties: collection.Seq[(String,Prop)] = props
+  def properties: collection.Seq[(String,Prop)] = {
+    frozen = true // once the properties have been exposed, they must be frozen
```

Comment:
an alternative would be to add a `private[scalacheck] def freeze(): Unit` and call it from the task when it gets the `properties` or right before execution